### ISSUE TITLE
Pressing tab doesn't move to the desired position anymore after upgrading from 1.13.1

### ISSUE THREAD

**Issue by cateyes99** [Microsoft/vscode#31015]
- VSCode Version: `1.14.1`
- OS Version: `OS X EI Capitan 10.11.3`

Steps to Reproduce:

1. I've some code in a C# file, create one **new `empty` line** in the middle of the code which has already a few indents, so the cursor can only be placed at the most-left position,  like the the Line 52 in the picture:
![image](https://user-images.githubusercontent.com/19678455/28362699-77de3f16-6cd1-11e7-977f-460461679dca.png)

2. Now press the <kbd>Tab</kbd>, it moves only **one** tab distance in this case which is annoying:
![image](https://user-images.githubusercontent.com/19678455/28362836-26971a96-6cd2-11e7-8428-fc378adb6f2c.png)
But with the previous version `1.13.1`, the same action will moves **three** tab distance as desired as shown below:
![image](https://user-images.githubusercontent.com/19678455/28362994-c1de4916-6cd2-11e7-98de-f3a5863616fc.png)

**Comment by cateyes99** [comment]
@rebornix i've downed the latest VS Code 1.15.0 insider: 
```
Version 1.15.0-insider (1.15.0-insider)

fed03b66ac8acdac93897727a1d19ee0f6898bf2

2017-07-25T07:15:46.026Z
```
But don't think the fix `6d7e8da` works well. It works only for the new line directly after a left curly brace `{`, like Line 7, but not Line 9, 11 as shown below:
![image](https://user-images.githubusercontent.com/19678455/28597243-e7f7eda0-71f0-11e7-951a-006c8f7a3929.png)

**Comment by cateyes99** [comment]
@rebornix this needs to be re-open. I don't think it has been fixed.
